### PR TITLE
refactor: Setup project for Typescript

### PR DESCRIPTION
**Pull request by redhaam**

### CODE REVIEW

**Comment by redhaam** [comment]
I needed to install `eslint-plugin-import` and `eslint-plugin-jest` since they aren't included in the eslint config even though it depends on them.

File: package.json [hunk 1]
```diff
@@ -50,13 +53,19 @@
...
   "devDependencies": {
-    "@babel/plugin-transform-modules-commonjs": "^7.16.0",
+    "@types/jest": "^27.0.2",
+    "@typescript-eslint/eslint-plugin": "^5.3.0",
+    "@typescript-eslint/parser": "^5.3.0",
     "eslint": "^8.1.0",
-    "eslint-config-cheminfo": "^7.2.0",
+    "eslint-config-cheminfo-typescript": "^10.2.2",
+    "eslint-plugin-import": "^2.25.2",
+    "eslint-plugin-jest": "^25.2.2",
```
Comment (targos) [inline package.json:62]: We expect npm7+ to be installed (it comes by default with Node.js 16 and 17). With it, the peer dependencies are installed automatically

File: package.json [hunk 2]
```diff
@@ -50,13 +53,19 @@
     "ml-matrix": "^6.8.0"
   },
   "devDependencies": {
-    "@babel/plugin-transform-modules-commonjs": "^7.16.0",
+    "@types/jest": "^27.0.2",
+    "@typescript-eslint/eslint-plugin": "^5.3.0",
+    "@typescript-eslint/parser": "^5.3.0",
```
Comment (targos) [inline package.json:58]: Same here.

**Review by targos — COMMENTED** [review]
I'm sorry we forgot to tell you that there were already basic type definitions in this project at https://github.com/mljs/pca/blob/master/pca.d.ts
Could you compare this file with the types you added (and fix them if necessary), then delete the file?

File: src/pca.ts [hunk 3]
```diff
@@ -12,9 +21,29 @@ import { Matrix, MatrixTransposeView, EVD, SVD, NIPALS } from 'ml-matrix';
...
+  public constructor(
+    dataset?: MaybeMatrix,
+    options: {
+      isCovarianceMatrix?: boolean;
+      method?: string;
+      nCompNIPALS?: number;
+      center?: boolean;
+      scale?: boolean;
+      ignoreZeroVariance?: boolean;
+    } = {},
```
Comment (targos) [inline src/pca.ts:43]: For objects, we like to define the types in interfaces with names. See https://github.com/mljs/pca/blob/master/pca.d.ts (ignore the `I` at the beginning of the interface names. It's a convention that we do not use anymore.)

Comment (targos) [reply src/pca.ts:43]: For example here, it should be:

```
export interface PCAOptions {
  isCovarianceMatrix?: boolean;
  method?: 'SVD' | 'NIPALS' | 'covarianceMatrix';
  center?: boolean;
  scale?: boolean;
  nCompNIPALS?: number;
  ignoreZeroVariance?: boolean;
}
```

```
options: PCAOptions = {},
```

File: src/__tests__/iris.ts [hunk 4]
```diff
@@ -44,6 +44,7 @@ describe('iris dataset test method covarianceMatrix', () => {
 
 describe('iris dataset test wrong method', () => {
   it('wrong method', () => {
+    // @ts-ignore
```
Comment (targos) [inline src/__tests__/iris.ts:47]:
```suggestion
    // @ts-expect-error
```

same for the other tests

**Comment by lpatiny** [comment]
The tests are currently failing. 

We could still keep the compatibility with node 12 and not use '?.' but simply check if the variable is undefined

**Pull request merged**